test/e2e/storage/drivers/in_tree.go

```diff
@@ -1772,9 +1774,11 @@ func (a *awsDriver) PrepareTest(f *framework.Framework) (*storageframework.PerTe
 
 func (a *awsDriver) CreateVolume(config *storageframework.PerTestConfig, volType storageframework.TestVolType) storageframework.TestVolume {
 	zone := getInlineVolumeZone(config.Framework)
```

Comment:
We'd want to rename the getInlineVolumeZone() ftn to getInlineOrPreprovisionedVolumeZone().

But maybe rather than adding a random dependency like this, it would be better to update GetPersistentVolumeSource for these driver to return a non-nil node affinity? eg [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L1335). Then the node affinity would be set up correctly in createPVCPV() if I read the code aright.

I think I'd prefer that latter approach given that it looks like we have to update a bunch of per-driver calls anyway. Also I think it's cleaner and more expected to do it when creating the PV rather than magically plumbing stuff through node selection, in addition to actually testing the common topology use case.

Reply:
+1 to fixing the volume node affinity. This change circumvents scheduler logic but adding volume node affinity will make sure that full scheduler logic is exercised when test runs. I know it is minor issue (and I hope we have tests elsewhere for scheduler but still.. :-) )